### ISSUE TITLE
Identical object gives nonempty diff

### ISSUE THREAD

**Issue by justinj** [andreyvit/json-diff#7]
Hi there,

nice tool here :)

I ran into (what I assume is) a small bug, shown by this example:

```
var jsondiff = require("json-diff");

var val = [{}];
console.log(jsondiff.diff(val, val)); // => [ [ '-', {} ], [ '+', {} ] ]
```

Which happens in the cli as well:

```
$ cat a.json        
[{}]
$ cat b.json 
[{}]
$ json-diff a.json b.json
 [
-  {
-  }
+  {
+  }
 ]
```

I understand if this is hard to fix based on the nature of diffing JSON in the first place, but it would be a nice improvement to have :)

I should mention that this actually occurred in a larger object for me, and this is just the simplest version I could find. I can provide the larger example too if that would be helpful.

**Comment by kanitw** [comment]
+1

**Comment by sirg** [comment]
Additional: 

 `[{"a":1}]` does NOT produce a spurious diff

`[{"a":1},{"a":1}]` also produces a spurious diff:
```
$ json-diff <(echo '[{"a":1},{"a":1}]') <(echo '[{"a":1},{"a":1}]')
 [
   ...
-  {
-    a: 1
-  }
+  {
+    a: 1
+  }
 ]
```

**Comment by ewoudenberg** [comment]
This has been fixed in 0.6.0:
```
$ json-diff --full  <(echo '[{}]') <(echo '[{}]')
 [
   {
   }
 ]
$ json-diff   <(echo '[{}]') <(echo '[{}]')
$
```
